workspace/__lib__/xod-dev/ssd1306-display/ssd1306-128x64-i2c-device/patch.cpp

```diff
@@ -0,0 +1,43 @@
+#pragma XOD error_raise enable
+
+// clang-format off
+{{#global}}
+#include <SSD1306.h>
+{{/global}}
+// clang-format on
+
+struct State {
+    uint8_t mem[sizeof(SSD1306)];
+    uint8_t* buffer;
+};
+
+using Type = SSD1306*;
+
+// clang-format off
+{{ GENERATED_CODE }}
+// clang-format on
+
+void evaluate(Context ctx) {
+    if (!isSettingUp())
+        return;
+
+    auto state = getState(ctx);
+
+    auto i2c = getValue<input_I2C>(ctx);
+    uint8_t addr = getValue<input_ADDR>(ctx);
+
+    if (addr > 127) {
+        raiseError(ctx);
+        return;
+    }
+
+    Type dev = new (state->mem) SSD1306(i2c, addr);
+
+    state->buffer = new uint8_t[dev->getScreenWidth() * dev->getScreenHeight() / 8];
```

Comment:
Take two :clapper: 

Do not use `new` to allocate the buffer. Keep it in `State`

```cpp
struct State {
    uint8_t mem[sizeof(SSD1306)];
    uint8_t buffer[128 * 64 / 8];
};
```